api/tests/Behat/bootstrap/v2/UserManagement/UserManagementTrait.php

```diff
@@ -0,0 +1,677 @@
+     */
+    public function iSearchForTestUsersWithTheAllFilter()
+    {
+        $this->searchUserWithFilter('', 'search-test-');
+        $this->setFixtureUserEmailsAndCount();
+    }
+
+    /**
+     * @When I search for one of the NDR test users with the All Roles filter
+     */
+    public function iSearchForTestUsersWithTheNDRFilter()
+    {
+        $this->iAmOnAdminUsersSearchPage();
+        $this->fillField('admin_q', 'search-test-');
+        $this->selectOption('admin[role_name]', '');
+        $this->checkOption('admin[ndr_enabled]');
+        $this->pressButton('Search');
+        $this->userCount = 1;
+        $this->userEmails = ['search-test-ndr-'.$this->testRunId.'@t.uk'];
+    }
+
+    private function setFixtureUserEmailsAndCount()
+    {
+        $this->userCount = 10;
+        $this->userEmails = [
+            'search-test-lay-'.$this->testRunId.'@t.uk',
+            'search-test-pa-n-'.$this->testRunId.'@t.uk',
+            'search-test-pa-'.$this->testRunId.'@t.uk',
+            'search-test-prof-n-'.$this->testRunId.'@t.uk',
+            'search-test-prof-'.$this->testRunId.'@t.uk',
+            'search-test-admin-'.$this->testRunId.'@t.uk',
+            'search-test-manager-'.$this->testRunId.'@t.uk',
+            'search-test-super-'.$this->testRunId.'@t.uk',
+            'search-test-ad-'.$this->testRunId.'@t.uk',
+            'search-test-ndr-'.$this->testRunId.'@t.uk',
+        ];
+    }
+
+    private function searchUserWithFilter(string $filterText, string $searchField)
+    {
+        $this->iAmOnAdminUsersSearchPage();
+        $this->fillField('admin_q', $searchField);
+        $this->selectOption('admin[role_name]', $filterText);
+        $this->pressButton('Search');
+    }
+
+    // ======= CREATE USER FUNCTIONALITY =======
+
+    /**
+     * @When I add a new lay deputy user
+     */
+    public function iAddNewLayDeputyUser()
+    {
+        $this->pressButton('Add new user');
+        $this->expectedUsers = [
+            [
+                'email' => 'add-user-1-'.$this->testRunId.'@t.com',
+                'firstName' => 'added',
+                'lastName' => 'user1',
+                'postcode' => 'B99 5ZZ',
+                'roleType' => 'deputy',
+                'role' => 'ROLE_LAY_DEPUTY',
+            ],
+        ];
+        $this->fillInAndSubmitUsers(
+            $this->expectedUsers[0]['email'],
+            $this->expectedUsers[0]['firstName'],
+            $this->expectedUsers[0]['lastName'],
+            $this->expectedUsers[0]['postcode'],
+            $this->expectedUsers[0]['roleType'],
+            $this->expectedUsers[0]['role']
+        );
+    }
+
+    /**
+     * @When I search for the newly created user
+     */
+    public function iSearchForNewlyCreatedUser()
+    {
+        $this->searchUserWithFilter('', $this->expectedUsers[0]['email']);
+    }
+
+    /**
+     * @When I can see the user as non active in search results
```

Comment:
Should this be using `Then` if it's an assertion?

Reply:
Yes! Good spot